### PR TITLE
PFS-30: Add debit

### PR DESCRIPTION
**Pull request by denialanderror**

### CODE REVIEW

File: finance-api/internal/store/queries/ledger_allocation.sql [hunk 1]
```diff
@@ -11,7 +11,6 @@ WITH filtered_ledger_allocation AS (
     from ledger l
              inner join ledger_allocation lc on lc.ledger_id = l.id
     where l.id = $1
-      and l.type IN ('CREDIT MEMO', 'CREDIT WRITE OFF')
```
Comment (josephsmith0705) [inline finance-api/internal/store/queries/ledger_allocation.sql:14]: Out of interest, why are we removing this where clause?

File: finance-api/internal/service/ledger_entries_test.go [hunk 2]
```diff
@@ -160,6 +160,58 @@ func TestService_ValidateAdjustmentAmount(t *testing.T) {
...
+			err: shared.BadRequest{Field: "Amount", Reason: "Amount entered must be equal to or less than £100"},
+		},
+		{
+			name: "Add Debit - valid",
+			adjustment: &shared.CreateLedgerEntryRequest{
+				AdjustmentType: shared.AdjustmentTypeAddDebit,
+				Amount:         22000,
+			},
+			balance: store.GetInvoiceBalanceRow{
+				Initial:     16000, // invoice calculated at £160 instead of £320 in error at creation
```
[the remark this answers is not in the document]
Comment (denialanderror) [reply finance-api/internal/service/ledger_entries_test.go:196]: It's covering a test scenario from the ticket. The comment doesn't add anything without that context though, so I'll remove it.

File: finance-api/internal/store/queries/ledger_allocation.sql (hunk 1 above)
Comment (denialanderror) [reply finance-api/internal/store/queries/ledger_allocation.sql:14]: We are already addressing the ledger allocation by id so the filter isn't necessary. I expect it was copy pasted from the fetch query, where it is needed.

Comment (denialanderror) [reply finance-api/internal/store/queries/ledger_allocation.sql:14]: I've added this back in (and refactored the query) as it is a useful guard against the query being used incorrectly.

**Review by josephsmith0705 — APPROVED** [review]

**Pull request merged**